File: python/30s/list/difference.py
"""
Returns the difference between two iterables.
Use list comprehension to only keep values not contained in b
"""


def difference(a, b):
    return [item for item in a if item not in b]


if __name__ == '__main__':
    print(difference([1,2,3], [2,3,4,5,6,7,8]))  # [1]

File: little-js/30s/array/difference.js
/*Returns the difference between two arrays.

Create a Set from b, then use Array.prototype.filter() on a to only keep values not contained in b.*/

const difference = (a, b) => {
  const s = new Set(b)
  return a.filter(x => !s.has(x))
}

console.log(difference([1, 2, 3], [1, 2, 3, 4, [6]]))
